src/database/schema/UserSchema.ts

```diff
@@ -305,11 +305,52 @@ class UserSchema implements BaseDbSchema {
         id: uuidv4(),
         channelId,
         userId: json?.user?.id,
-        username: json?.user?.name,
+        username: json?.user?.username,
         countryCode: '',
         createdAt: json?.user?.created_at,
         updatedAt: json?.updated_at,
         lastActiveAt: json?.user?.last_active,
+        profilePicture: json?.user?.image || DEFAULT_PROFILE_PIC_PATH,
+        bio: '',
+        isBanned: json?.banned,
+        anon_user_info_color_code: json?.anon_user_info_color_code,
+        anon_user_info_color_name: json?.anon_user_info_color_name,
+        anon_user_info_emoji_code: json?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: json?.anon_user_info_emoji_name
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is trying to access properties like `username`, `created_at`, `last_active`, `image`, and `banned` directly from `json?.user`. However, it's not clear if these properties always exist in the `user` object. If any of these properties are missing in the `user` object, it could potentially lead to undefined values being set in the `UserSchema`. It would be better to add checks for these properties before accessing them.

```diff
- username: json?.user?.username,
+ username: json?.user && json.user.hasOwnProperty('username') ? json.user.username : null,

- createdAt: json?.user?.created_at,
+ createdAt: json?.user && json.user.hasOwnProperty('created_at') ? json.user.created_at : null,

- lastActiveAt: json?.user?.last_active,
+ lastActiveAt: json?.user && json.user.hasOwnProperty('last_active') ? json.user.last_active : null,

- profilePicture: json?.user?.image || DEFAULT_PROFILE_PIC_PATH,
+ profilePicture: json?.user && json.user.hasOwnProperty('image') ? json.user.image : DEFAULT_PROFILE_PIC_PATH,

- isBanned: json?.banned,
+ isBanned: json.hasOwnProperty('banned') ? json.banned : null,
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->